### DISCUSSION TITLE
Tool to analyse KV store size over time?

### DISCUSSION THREAD

**Discussion by letmaik** [microsoft/CCF#3704] [Ideas]
There's the `ccf.ledger_viz` tool which gives per-transaction information about the transaction type. Sometimes, when debugging memory-related issues, it would be useful to know whether or how much the KV store is growing over time. Does this make sense?

**Comment by jumaffre** [comment]
Yep, this was hinted at before (e.g. with regards to https://github.com/microsoft/CCF/issues/1858) and has been on my nice-to-have list for a while now, but other items have taken priority. The existing [`GET /node/memory`](https://microsoft.github.io/CCF/main/operations/operator_rpc_api.html#get--node-memory) endpoint is a pretty good approximation of the current size of the store as we do not buffer much data inside the enclave memory (although this probably isn't as true anymore with indexing, so maybe this feature becomes more useful).

**Comment by eddyashton** [comment]
Regular snapshots will also give you an indication of KV size over time. We could track the size of the public tables using the Python ledger parser, but this can't tell us anything about private state (if every transaction contains 1MB of private state, we don't know if that's a _new_ 1MB or overwriting a previous 1MB).
